internal/registry/manifest.go

```diff
@@ -344,11 +350,17 @@ func (handler *manifests) handleGet(resp http.ResponseWriter, req *http.Request,
 	if _, err := io.Copy(resp, &buf); err != nil {
 		return regErrInternal(err)
 	}
+	if err := handler.audit.AuditPull(ctx, repo, target); err != nil {
```

Comment:
Bug: This is never reached when redirects are enabled (which they are by default). Please also add calling of `AuditPull` above, where the `RedirectError` is handled.

After trying this, there is another error: 
> could not query ArtifactVersion: too many rows in result set

I was able to fix this by changing `db.GetDigestArtifactVersion` to use `SELECT DISTINCT`.